modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4Transport.java

```diff
@@ -301,6 +300,7 @@ protected void initChannel(Channel ch) throws Exception {
             addClosedExceptionLogger(ch);
             assert ch instanceof Netty4NioSocketChannel;
             NetUtils.tryEnsureReasonableKeepAliveConfig(((Netty4NioSocketChannel) ch).javaChannel());
+            ch.pipeline().addLast("byte_buf_sizer", NettyByteBufSizer.INSTANCE);
```

Comment:
I think this will go before the ssl handling which I remember as possibly leading to an extra/unnecessary copy?

Reply:
I don't think it will, we explicitly add the ssl handler via `addFirst` here:

https://github.com/elastic/elasticsearch/blob/a92a647b9f17d1bddf5c707490a19482c273eda3/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/transport/netty4/SecurityNetty4Transport.java#L166

and then replace that first handler here:

https://github.com/elastic/elasticsearch/blob/a92a647b9f17d1bddf5c707490a19482c273eda3/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/transport/netty4/SecurityNetty4Transport.java#L207

so this will always be after the SSL handler I think.

Reply:
Sorry for overlooking that, thanks for pointing it out!